pkg/storage/etcd3/watcher.go

```diff
@@ -158,13 +158,17 @@ func (wc *watchChan) sync() error {
 	wc.initialRev = getResp.Header.Revision
 
 	for _, kv := range getResp.Kvs {
-		prevResp, err := wc.watcher.client.Get(wc.ctx, string(kv.Key), clientv3.WithRev(kv.ModRevision-1), clientv3.WithSerializable())
-		if err != nil {
-			return err
-		}
 		var prevVal []byte
-		if len(prevResp.Kvs) > 0 {
-			prevVal = prevResp.Kvs[0].Value
+		if kv.ModRevision != kv.CreateRevision {
+			prevResp, err := wc.watcher.client.Get(wc.ctx, string(kv.Key), clientv3.WithRev(kv.ModRevision-1), clientv3.WithSerializable())
+			if err != etcdrpc.ErrCompacted {
+				if err != nil {
+					return err
+				}
+				if len(prevResp.Kvs) > 0 {
+					prevVal = prevResp.Kvs[0].Value
+				}
+			}
```

Comment:
I though that we agree that we would like to be fully compatible with etcd2 (whith means that will send "Add" watch event for all objects that exists (and not modify for some of them). Pointers to etcd2 code:
https://github.com/kubernetes/kubernetes/blob/master/pkg/storage/etcd/etcd_watcher.go#L253
https://github.com/kubernetes/kubernetes/blob/master/pkg/storage/etcd/etcd_watcher.go#L457

With the code above, for some of them we will return "Add" and for some "Modify" (depending whether the information was compacted or not) - which in my opinion is even more misleading.
Why not just always pass empty prevVal to sendEvent?